### PR TITLE
remove arcs directory

### PR DESCRIPTION
**Pull request by harsha-mandadi**

### CODE REVIEW

File: src/ir/proto/raksha_ir.proto
Comment (bgogul) [inline src/ir/proto/raksha_ir.proto:130]: Can we split this off to a separate proto file  (e.g., type.proto) instead of adding this to raksha_ir.proto?

File: src/ir/proto/raksha_ir.proto
Comment (bgogul) [inline src/ir/proto/raksha_ir.proto:288]: Remove this as it is not needed.

File: src/analysis/souffle/tests/arcs_manifest_tests_todo/BUILD
Comment (markww) [inline src/analysis/souffle/tests/arcs_manifest_tests_todo/BUILD:17]: This whole `arcs_manifest_tests_todo` directory can be deleted. It contains arcs manifests that we eventually wanted to turn into tests. But since the arcs frontend is no longer being supported, there's no reason to keep them around.

File: src/ir/proto/raksha_type.proto
Comment (markww) [inline src/ir/proto/raksha_type.proto:25]: I'm not sure if we should even keep around this `TypeProto` file. While we are making some minimal use of encoding and decoding these protos in Raksha types, we aren't actually using types for anything in Raksha at the moment. It might be best to just have a clean break from the Arcs conception of types and build our own types as we need them. Otherwise, we risk sleepwalking into following the Arcs path, which might not always work for us.

For instance, does it make sense to think of a `BIGINT` as a primitive type? Or a `DURATION`? This is one of those things where the answer is "maybe, maybe not". In both the Souffle and potential custom solver case, the only primitive types (in this case meaning those that map directly to a simple type in the underlying runtime) would be `string`, `int64_t`, `uint64_t`, and `double`.

File: src/ir/types/proto/entity_type.h
Comment (markww) [inline src/ir/types/proto/entity_type.h:25]: I think we might want to get rid of `EntityType`s and `Schema`s altogether. We aren't using them in any significant way, and we might want to build something different for Raksha's purposes.

Comment (bgogul) [reply src/ir/types/proto/entity_type.h:25]: Let us keep them around for now. This would become useful/needed when we start talking about SQL tables and their schemas.

File: src/ir/proto/raksha_type.proto
Comment (bgogul) [reply src/ir/proto/raksha_type.proto:25]: I think it might be a good idea to simplify the proto to just the things we use currently.

**Review by bgogul — COMMENTED** [review]
Marked some parts of the proto for removal. PTAL.

File: src/ir/types/proto/entity_type.h
Comment (harsha-mandadi) [reply src/ir/types/proto/entity_type.h:25]: Acknowledge

**Review by bgogul — APPROVED** [review]
Thank you!

**Review by markww — APPROVED** [review]
LGTM modulo getting rid of the last node dependencies.

**Comment by markww** [comment]
Also, be sure to set `ready to pull` on this to verify that there's no internal footprint for this change before merging. This has the potential to be really painful to undo, so best to be careful.

File: gcb/presubmit.sh
Comment (harsha-mandadi) [reply gcb/presubmit.sh:99]: //src/analysis/souffle/examples/... are not dependent on arcs in any way. We removed the rust/arcs dependency and use raksha_policy_verifier instead. Why do you think they should go?

Comment (markww) [reply gcb/presubmit.sh:99]: OK, if they don't depend on Arcs, we should keep them.

**Pull request closed without merging**
